Source/WebKit/UIProcess/WebAuthentication/fido/CtapAuthenticator.cpp

```diff
@@ -128,6 +135,16 @@ void CtapAuthenticator::continueMakeCredentialAfterResponseReceived(Vector<uint8
             receiveRespond(ExceptionData { InvalidStateError, "At least one credential matches an entry of the excludeCredentials list in the authenticator."_s });
             return;
         }
+        if (error == CtapDeviceResponseCode::kCtap2ErrKeyStoreFull) {
+            auto& options = std::get<PublicKeyCredentialCreationOptions>(requestData().options);
+            if (options.authenticatorSelection->residentKey || options.authenticatorSelection->residentKey == ResidentKeyRequirement::Required) {
```

Comment:
Is this logic correct? When would you have a `options.authenticatorSelection->residentKey` that evaluates to `false`, but is also holding a value of `ResidentKeyRequirement::Required`?

Reply:
Should the first part of that conditional be `options.authenticatorSelection->requireResidentKey`, instead?